### PR TITLE
feat: add coverage scripts

### LINKED ISSUE #15003: Feature Request: Adding a script to get code coverage results
### Feature Description

Currently, there is no direct way for developers to get the test coverage results from the `go` CLI in an easy UI (HTML-based). 

The current way to do the same is:
- Run `go test -coverprofile=test.out <name of the package you want to test>`
- Install the `cmd` tool provided by Go by `go get golang.org/x/tools/cmd/cover`
- Generate the web UI for test results by running `go tool cover -html=test.out`

This is cumbersome and makes it difficult to check for coverage. 

## Proposal
Add `golang.org/x/tools/cmd/cover` to the project, and create a `make` script that runs the test on the package you specify and then uses the `cover` tool to show the coverage to users. 

### Use Case(s)

Any developers adding or changing functionalities can use the same to grasp the extent of their test coverage for a particular package. Also, for developers writing tests, this would allow to same a UI that shows exactly which lines aren't covered, so it becomes easier to target them and write tests for them.

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Description

Edited the `Makefile` to add three new scripts:
- `coverage`: This target runs the tests and generates a coverage report in the coverage.out file. It utilizes the go test command with the -coverprofile flag to collect coverage data. 
It can be run as `make PACKAGE=<path to go package> coverage`
After running the same, the browser shows an HTML page indicating the current coverage. 
image

- `coverage-clean`: This target removes the generated coverage file (`coverage.out`). It's useful for cleaning up after running coverage tests.
- `coverage-help`: This target displays a usage message with information about available options and commands related to coverage. 

## Related Issue(s)
Fixes #15003 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by VaibhavMalik4187** [comment]
I use the following alias to check to code coverage of any go package:
`alias check_coverage="go test ./... -coverprofile=cover.out && go tool cover -html=cover.out -o coverage.html && rm cover.out && firefox coverage.html"`
It is very flexible and can be used in any repository and in any directory without specifying the package name. I wonder if suggesting contributors to use a similar alias would be a better idea.

**Comment by EshaanAgg** [comment]
Seems cool, but how would you customize the package that you want to test and see the reports for with this alias? Running the tests for all the packages every time is not feasible.

File: Makefile [hunk 1]
```diff
@@ -423,3 +423,23 @@ generate-flag-testdata:
...
+
+
+
+# Scripts related to test coverage
+PACKAGE ?= ./go/...
+
+coverage:
+	go install golang.org/x/tools/cmd/cover
+	go test -coverprofile=coverage.out $(PACKAGE)
+	go tool cover -html=coverage.out
```
Comment (frouioui) [inline Makefile:435]: This feels very similar to the already existing `unit_test_cover` Makefile target. The following snippet is going to be important is testing the `sqlparser` package:
```sh

# Handle go tool cover failures due to not handling `//line` directives, which
# the goyacc compiler adds to the generated parser in sql.go. See:
# https://github.com/golang/go/issues/41222
sed -i'' -e '/^vitess.io\/vitess\/go\/vt\/sqlparser\/yaccpar/d' coverage.out
```

Comment (EshaanAgg) [reply Makefile:435]: Ahh got it. Should I rename the same to `test.coverage` to avoid the target conflict?

File: Makefile [hunk 2]
```diff
@@ -423,3 +423,23 @@ generate-flag-testdata:
...
 	./tools/get_kubectl_kind.sh
+
+
+
+# Scripts related to test coverage
+PACKAGE ?= ./go/...
+
+coverage:
+	go install golang.org/x/tools/cmd/cover
+	go test -coverprofile=coverage.out $(PACKAGE)
```
Comment (Maniktherana) [inline Makefile:434]: Just a thought, but I often find myself using the `-v` flag too as it helps find failing tests. Might be worth adding

**Pull request closed without merging**